### PR TITLE
Mixins Scss

### PR DESCRIPTION
**Pull request by HoseinEmrani**
- `mixins.scss` added into project:

> - shorthand for positioning,
> - breakpoint for responsive developing
> - shorthand for buttons styling
- improvement on sass codes in `button.scss`

### CODE REVIEW

File: examples/helloworld.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <meta name="author" content="Afshin Mehrabani (afshin.meh@gmail.com)">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link href='//fonts.googleapis.com/css?family=Open+Sans:400,300,600,700' rel='stylesheet' type='text/css'>
-  <link rel="stylesheet" href="../css/kissui.css">
+  <link rel="stylesheet" href="../docs/css/kissui.css">
```
Comment (afshinm) [inline examples/helloworld.html:10]: Sorry didn't get the reason of this change.

File: examples/helloworld_menu.html [hunk 2]
```diff
@@ -7,7 +7,7 @@
   <meta name="author" content="Afshin Mehrabani (afshin.meh@gmail.com)">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link href='//fonts.googleapis.com/css?family=Open+Sans:400,300,600,700' rel='stylesheet' type='text/css'>
-  <link rel="stylesheet" href="../css/kissui.css">
+  <link rel="stylesheet" href="../docs/css/kissui.css">
```
Comment (afshinm) [inline examples/helloworld_menu.html:10]: and this

File: scss/base/_variables.scss [hunk 3]
```diff
@@ -13,6 +13,18 @@ $bp-smaller-than-tablet    : "max-width: 750px" !default;
 $bp-smaller-than-desktop   : "max-width: 1000px" !default;
 $bp-smaller-than-desktophd : "max-width: 1200px" !default;
 
+// Breakpoints variable for breakpoint mixin
+$larger-than-mobile    : 400px !default;
```
Comment (afshinm) [inline scss/base/_variables.scss:17]: this change seems to be redundant as we already have breakpoints: https://github.com/HoseinEmrani/kissui/blob/c4e4e71aa2b7a881964b36972fb09fdab3ad3869/scss/base/_variables.scss#L5

File: scss/modules/_buttons.scss [hunk 4]
```diff
@@ -7,19 +7,37 @@ button {
   height: 38px;
   padding: 0 30px;
   color: $secondary-color;
-  text-align: center;
-  font-size: 11px;
-  font-weight: 600;
+  font: {
```
Comment (afshinm) [inline scss/modules/_buttons.scss:10]: nice one 👍

File: scss/modules/_buttons.scss [hunk 5]
```diff
@@ -30,83 +48,54 @@ input {
...
-  text-decoration: none;
-}
+    &:hover {
+      color: $dark-grey;
+      border-color: lighten($dark-grey, 33.3%);
+      outline: 0;
+    }
 
-input {
-  &[type="submit"]:hover,
```
Comment (afshinm) [inline scss/modules/_buttons.scss:58]: why did you remove `submit`, `reset` and `button` types?

**Comment by afshinm** [comment]
cool. I added some notes, please answer them to push this forward.

File: examples/helloworld.html (hunk 1 above)
Comment (HoseinEmrani) [inline examples/helloworld.html:10]: Oh! this change was for first time i cloned the repo, `kissui.css` wasn't linked to examples. forget to undo my changes.

File: scss/base/_variables.scss (hunk 3 above)
Comment (HoseinEmrani) [inline scss/base/_variables.scss:17]: breakpoints we already have are only usable in this way: 
`@media ( #{$breakpoint} ) {
// some styles
}`
but this breakpoints are for `breakpoint` mixin, developer  can use this mixin for inline css responsive. for example: 

``` scss
.first-class {
     .second-class {
        //some styles
       @include breakpoint($breakpoint) {
           // responsive styles
       }
    }
} 
```

will compile to: 

``` css
.first-class .second-class { // some styles }
@media (max-width: $breakpoint) {
 .first-class .second-class { // responsive styles } 
}
```

The mixin has two attributes. `breakpoint($max-width, $min-width)` min-width is optional.

File: scss/modules/_buttons.scss (hunk 5 above)
Comment (HoseinEmrani) [inline scss/modules/_buttons.scss]: It was my fault. I just used the sass inheritance but I forget to add them again to the root element. sorry!

**Comment by HoseinEmrani** [comment]
I think it's better to close this pull request and I'll send another one soon.

File: scss/modules/_buttons.scss (hunk 5 above)
Comment (HoseinEmrani) [inline scss/modules/_buttons.scss:58]: Just reviewed the code and find that they are at line 44 of this file.
